tests/test_event_processor.py

```diff
@@ -494,6 +494,34 @@ def on_log_event(log_event):
             1, len(self.optimizely.notification_center.notification_listeners[enums.NotificationTypes.LOG_EVENT]),
         )
 
+    def test_warning_log_level_on_queue_overflow(self):
+        """ Test that a warning log is created when events overflow the queue. """
+
+        # create scenario where the batch size (MAX_BATCH_SIZE=10) is larger than the queue size (7)
+        test_max_queue_size = 7
+
+        event_dispatcher = CustomEventDispatcher()
+
+        with mock.patch.object(self.optimizely, 'logger') as mock_config_logging:
+            self.event_processor = BatchEventProcessor(
+                event_dispatcher,
+                self.optimizely.logger,
+                True,
+                queue.Queue(maxsize=test_max_queue_size),
+            )
+
+        for i in range(0, self.MAX_BATCH_SIZE):
+            user_event = self._build_conversion_event(self.event_name)
+            self.event_processor.process(user_event)
+            event_dispatcher.expect_conversion(self.event_name, self.test_user_id)
+
+        time.sleep(self.TEST_TIMEOUT)
```

Comment:
Have you tried increasing the TEST_TIMEOUT by chance? It seems that the timings can behave differently in some versions of pytyhon.

Reply:
@The-inside-man to fix the flakiness? Yes, trying that now. 
Locally 4 our of 20 would fail, but with adjusted timeout and smllaer batch size I get 1/20
The other issue is Py 3.4 and PyPy/PyPy3 failing. Will see if adjusted timeout/batch size fix it

Reply:
Sounds good! Let me know if you need another set of eyes or anything.